src/NumericalAlgorithms/DiscontinuousGalerkin/MetricIdentityJacobian.hpp

```diff
@@ -0,0 +1,140 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+/// \cond
+class DataVector;
+template <size_t Dim>
+class Mesh;
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace dg {
+/*!
+ * \brief Compute the Jacobian determinant times the inverse Jacobian so that
+ * the result is divergence-free.
```

Comment:
I suggest using (and perhaps explaining) the phrase "metric identity-satisfying Jacobian" so that the name of the function/file is easier to relate to the code.

Perhaps the phrase can be used in the brief, or if that's too bulky for the brief, then it can be used near the top of the main documentation. Then, perhaps at the bottom where the identify is given, you can restate the name by way of explanation? These are just my first ideas, but you've thought about this more so it's totally up to you how you'd like to structure it :)